compiler/src/dotty/tools/backend/jvm/GenBCode.scala

```diff
@@ -214,7 +214,8 @@ class GenBCodePipeline(val entryPoints: List[Symbol], val int: DottyBackendInter
             } else if (!outTastyFile.isVirtual) {
               // Create an empty file to signal that a tasty section exist in the corresponding .class
               // This is much cheaper and simpler to check than doing classfile parsing
-              outTastyFile.create()
+              val hasTasty = getFileForClassfile(outF, store.name, ".hasTasty")
```

Comment:
It turns out that `getFileForClassfile` will automatically create an empty file, so we are now creating both ".tasty" and ".hasTasty" files, I'll fix it :).